staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -61,42 +61,56 @@ func (w *realTimeoutFactory) TimeoutCh() (<-chan time.Time, func() bool) {
 	return t.C, t.Stop
 }
 
-// serveWatch handles serving requests to the server
+// serveWatch will serve a watch response.
 // TODO: the functionality in this method and in WatchServer.Serve is not cleanly decoupled.
-func serveWatch(watcher watch.Interface, scope RequestScope, req *http.Request, w http.ResponseWriter, timeout time.Duration) {
-	// negotiate for the stream serializer
-	serializer, err := negotiation.NegotiateOutputStreamSerializer(req, scope.Serializer)
+func serveWatch(watcher watch.Interface, scope RequestScope, mediaTypeOptions negotiation.MediaTypeOptions, req *http.Request, w http.ResponseWriter, timeout time.Duration) {
```

Comment:
there's a lot of stuff in mediaTypeOptions, and we're ignoring almost all of it. we pull media type from serializer like this:
```
// TODO: next step, get back mediaTypeOptions from negotiate and return the exact value here
mediaType := serializer.MediaType
if mediaType != runtime.ContentTypeJSON {
	mediaType += ";stream=watch"
}
```
instead of using `mediaTypeOptions.Accepted`

we also don't check if `mediaTypeOptions.Accepted.Serializer` is set

is there a way to get the information we need more crisply?

Reply:
Serializer is always set - it's not allowed to be. nil.

Reply:
And this doesn't change the negotiation that we had before (in terms of behavior relative to 1.13).